docs/4-Prototyping.md

```diff
@@ -1,29 +1,172 @@
 # Prototyping
-## Overview
+
+A website prototype can be any mock-up or demo of what a website could look like when it goes live. A living prototype is a prototype that's actively used throughout the lifecycle of the website. Even after the site is deployed, the prototype lives on as an active demonstration of what the website looks like.
+
+The following features and conventions come together to allow the creation of a living prototype.
+
+## Conventions
+
+### Meta-Data
+
+Meta-Data is general data that helps describe the content you're presenting. Depending on the scope, meta-data can exist almost anywhere.
+
+#### Local
+
+Add YAML front-matter to files to append properties to the file.
+
+*example.html*
+``` html
+---
+name: Linus Torvalds
+languages:
+- PHP
+- Perl
+- Python
+---
+<h1>Linus Torvalds</h1>
+```
+
+*output*
+``` html
+<h1>Linux Torvalds</h1>
+```
+
+#### Global
+
+You can add global meta-data through the use of the [metalsmith-metadata-convention](https://www.npmjs.com/package/metalsmith-metadata-convention) plugin and `.metadata` files.
+
+*authors.metadata*
+``` yaml
+---
+Stephen King:
+  birthdate: 1947
+J. K. Rowling:
+  birthdate: 1965
+William Shakespeare:
+  birthdate: 1564
+Nora Roberts:
+  birthdate: 1950
+---
+
+This list of authors is loaded into a global metadata variable named "authors".
+```
+
+### Transformers
+
+Naming a file by a [JSTransformer](https://github.com/jstransformers/jstransformer) name will have it process with the engine. The following is an example of using [Twig](http://twig.sensiolabs.org):
```

Comment:
@RobLoach what's the point of transformers? Is it to enable you to use other templating engines? I think an explanation of what Kalastatic uses transformers for would be useful.